cmd/argocd/commands/app.go

```diff
@@ -1451,6 +1479,24 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 						},
 					}
 				}
+				if diffChanges {
+					app, err := appIf.Get(context.Background(), &applicationpkg.ApplicationQuery{Name: &appName})
+					errors.CheckError(err)
+					resources, err := appIf.ManagedResources(context.Background(), &applicationpkg.ResourcesQuery{ApplicationName: &appName})
+					errors.CheckError(err)
+					conn, settingsIf := acdClient.NewSettingsClientOrDie()
+					defer argoio.Close(conn)
+					argoSettings, err := settingsIf.Get(context.Background(), &settingspkg.SettingsQuery{})
+					errors.CheckError(err)
+					foundDiffs := false
+					findandPrintDiff(app, resources, argoSettings, appName, &foundDiffs, diffOption)
```

Comment:
I think this will crash with a nil-pointer dereference if `--local` is not specified, because `diffOption` will be nil (it's only initialized above if `local == true`).

Reply:
It would not crash as have initialized diffoption  for each appname passed irrespective of flag and default value for member of structs is "". i'll still validate